### PR TITLE
Pika - add publish_hook and consume_hook

### PR DESCRIPTION
**Pull request by ItayGibel-helios**

### CODE REVIEW

File: opentelemetry-instrumentation/src/opentelemetry/instrumentation/bootstrap_gen.py [hunk 1]
```diff
@@ -18,125 +18,125 @@
...
+        "instrumentation": "opentelemetry-instrumentation-urllib3==0.25b1",
     },
 }
 default_instrumentations = [
-    "opentelemetry-instrumentation-dbapi==0.25b0",
-    "opentelemetry-instrumentation-logging==0.25b0",
-    "opentelemetry-instrumentation-sqlite3==0.25b0",
-    "opentelemetry-instrumentation-urllib==0.25b0",
-    "opentelemetry-instrumentation-wsgi==0.25b0",
+    "opentelemetry-instrumentation-dbapi==0.25b1",
```
Comment (nozik) [inline opentelemetry-instrumentation/src/opentelemetry/instrumentation/bootstrap_gen.py]: I wouldn't make these updates upstream

File: instrumentation/opentelemetry-instrumentation-pika/tests/test_utils.py [hunk 2]
```diff
@@ -284,3 +328,51 @@ def test_decorate_basic_publish_no_properties(
...
+        inject: mock.MagicMock,
+        get_span: mock.MagicMock,
+    ) -> None:
+        callback = mock.MagicMock()
+        tracer = mock.MagicMock()
+        channel = mock.MagicMock(spec=Channel)
+        method = mock.MagicMock(spec=Basic.Deliver)
+        properties = mock.MagicMock()
+        mock_body = b"mock_body"
+        publish_hook = mock.MagicMock()
```
Comment (nozik) [inline instrumentation/opentelemetry-instrumentation-pika/tests/test_utils.py:376]: Can you add a test that publish hook is a real function and not a mock?

File: instrumentation/opentelemetry-instrumentation-pika/tests/test_utils.py [hunk 3]
```diff
@@ -211,6 +211,50 @@ def test_decorate_callback(
...
+        get_span: mock.MagicMock,
+    ) -> None:
+        callback = mock.MagicMock()
+        mock_task_name = "mock_task_name"
+        tracer = mock.MagicMock()
+        channel = mock.MagicMock(spec=Channel)
+        method = mock.MagicMock(spec=Basic.Deliver)
+        properties = mock.MagicMock()
+        mock_body = b"mock_body"
+        consume_hook = mock.MagicMock()
```
Comment (nozik) [inline instrumentation/opentelemetry-instrumentation-pika/tests/test_utils.py:232]: Same for consume hook - could you do that with a real function?

**Review by nozik — CHANGES_REQUESTED** [review]

File: opentelemetry-instrumentation/src/opentelemetry/instrumentation/bootstrap_gen.py (hunk 1 above)
Comment (ItayGibel-helios) [reply opentelemetry-instrumentation/src/opentelemetry/instrumentation/bootstrap_gen.py]: this file is auto-generated

**Review by nozik — APPROVED** [review]

**Pull request closed without merging**